### PR TITLE
finish

### PR DESCRIPTION
**Pull request by daothanhcam**

### CODE REVIEW

File: app/views/admin/users/index.html.erb [hunk 1]
```diff
@@ -3,6 +3,10 @@
   <h1>All users<br>
```
Comment (phonghenry14) [inline app/views/admin/users/index.html.erb]: i18n

File: app/controllers/admin/users_controller.rb [hunk 2]
```diff
@@ -1,11 +1,27 @@
...
+  before_action :set_user, except: [:index, :new, :create]
 
   def index
     @users = User.paginate page: params[:page]
   end
 
+  def new
+    @user = User.new
+    @position = Position.all
+    @user.position_users.build
```
Comment (khanhkhanh) [inline app/controllers/admin/users_controller.rb]: Try removing this line

File: app/controllers/admin/users_controller.rb [hunk 3]
```diff
@@ -1,11 +1,27 @@
...
-  before_action :set_user, except: :index
+  before_action :set_user, except: [:index, :new, :create]
 
   def index
     @users = User.paginate page: params[:page]
   end
 
+  def new
+    @user = User.new
+    @position = Position.all
```
Comment (khanhkhanh) [inline app/controllers/admin/users_controller.rb]: @positions instead of @position

File: app/views/admin/users/_form_user [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+      <%= f.label t "user.password_confirmation" %><br />
+      <%= f.password_field :password_confirmation, autocomplete: "off",
+        class: "form-control" %>
+
+      <%= f.label t "user.birthday" %>
+      <%= f.datetime_field :birthday, class: "form-control" %><br />
+
+      <%= f.fields_for :position_users do |p| %>
+        <%= p.label :position_id, "Position" %>
```
Comment (khanhkhanh) [inline app/views/admin/users/_form_user]: Remove "Position" or use i18n

File: app/views/admin/users/index.html.erb (hunk 1 above)
Comment (khanhkhanh) [inline app/views/admin/users/index.html.erb]: i18n

File: app/controllers/admin/users_controller.rb [hunk 5]
```diff
@@ -31,8 +46,9 @@ def set_user
     @user = User.find params[:id]
   end
 
-  def params_user
-    params.require(:user)
-      .permit :name, :email, :password, :password_confirmation, :role
+  def user_params
+    params.require(:user).permit :username, :email, :role, :password,
+      :password_confirmation, :birthday,
+      position_users_attributes: [:id, :user_id, :position_id]
```
Comment (dieunb) [inline app/controllers/admin/users_controller.rb]: `_destroy` ?

**Pull request merged**